word_language_model/main.py

```diff
@@ -130,14 +157,24 @@ def evaluate(data_source):
     model.eval()
     total_loss = 0.
     ntokens = len(corpus.dictionary)
-    hidden = model.init_hidden(eval_batch_size)
+    if args.model == 'Transformer':
+        tgt_mask = model.transformer.generate_square_subsequent_mask(args.bptt).to(device)
+    else:
+        hidden = model.init_hidden(eval_batch_size)
     with torch.no_grad():
         for i in range(0, data_source.size(0) - 1, args.bptt):
             data, targets = get_batch(data_source, i)
-            output, hidden = model(data, hidden)
+
+            if args.model == 'Transformer':
+                if args.bptt > len(data_source) - 1 - i:
```

Comment:
See if you can remove this, because it's throwing data away.

Reply:
It will not throwing data away. If the data left has a length short then args.bptt, it has to generate a new tgt_mask.